PowerPointLabs/PowerPointLabs/Utils/FileDir.cs

```diff
@@ -139,6 +139,16 @@ public static bool MoveFolder(string oldPath, string newPath)
             return CopyFolder(oldPath, newPath) && DeleteFolder(oldPath);
         }
 
+        public static string GetTemporaryPngFilePath()
+        {
+            string tempDirectory = Path.GetTempPath();
+            Directory.CreateDirectory(tempDirectory);

```

Comment:
Not sure if it is possible, but could an exception be caught if there is insufficient permissions to create a directory or a new file? There could be PPTLabs users that are in more protected environments or maybe they did not run powerpoint with administrator permissions.

Reply:
Thus if they do not have sufficient permissions, copying and pasting actually works better for them.

Reply:
Good point, I didn't think about that! Ok, I have added a try catch for exceptions as a result of creating/deleting files. The execution then proceeds with copy and paste.

Reply:
I only thought about that because of the numerous bug reports filed regarding insufficient permissions on google groups 🔢